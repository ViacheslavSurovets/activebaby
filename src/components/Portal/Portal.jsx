import React, { useEffect, useMemo } from 'react';
import { createPortal } from 'react-dom';

const Portal = ( { component: ComponentPortal } ) => {
  const el = useMemo ( () =>(
    document.createElement ( 'div' )
), [] );
  el.setAttribute( 'id', 'portal' );
  useEffect ( () => {
    document.body.appendChild ( el );
    return () => {
      document.body.removeChild ( el );
    };
  }, [ el ] );

  return (
    createPortal ( <ComponentPortal /> , el )
  );
};

export default Portal;
