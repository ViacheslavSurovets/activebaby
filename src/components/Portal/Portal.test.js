import React from 'react';
import { shallow } from 'enzyme';
import Portal from './Portal';


describe ( 'Portal', () => {
    const MockComponent = () => (
      <h1>Mock Component</h1>
    );

  it ( 'Portal Snapshot', () => {
    expect ( shallow ( <Portal component={ MockComponent } /> ) ).toMatchSnapshot ();
  } );
} );
