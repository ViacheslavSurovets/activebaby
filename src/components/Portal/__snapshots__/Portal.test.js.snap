// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal Portal Snapshot 1`] = `ShallowWrapper {}`;
